src/Component/src/Grid/State/RequestGridProvider.php

```diff
@@ -64,8 +64,32 @@ public function provide(Operation $operation, Context $context): object|array|nu
         if ($data instanceof Pagerfanta) {
             $currentPage = $request->query->getInt('page', 1);
             $data->setCurrentPage($currentPage);
+
+            $maxPerPage = $this->resolveMaxPerPage(
+                $request->query->has('limit') ? $request->query->getInt('limit') : null,
+                $gridDefinition->getLimits(),
+            );
+            $data->setMaxPerPage($maxPerPage);
         }
 
         return $gridView;
     }
+
+    private function resolveMaxPerPage(?int $requestLimit, array $gridLimits = []): int
+    {
+        if (null === $requestLimit) {
+            $firstGridLimit = reset($gridLimits);
+
+            return false === $firstGridLimit ? 10 : $firstGridLimit;
```

Comment:
`10` over here feels like a magical number.

It would be better to have fallback limit (10) at grid definition or at least make `10` a const on this class.

Reply:
Yeah, indeed, for now I've added a simple constant on this class.